src/Aspire.Dashboard/Components/Controls/EnvironmentVariables.razor

```diff
@@ -8,6 +8,7 @@
         @if (ShowSpecOnlyToggle)
         {
             <FluentButton Appearance="Appearance.Lightweight"
+                          Disabled="IsSpecOnlyToggleDisabled"
                           IconEnd="@(_showAll ? _showSpecOnlyIcon : _showAllIcon)"
                           Title="@(_showAll ? Loc[ControlsStrings.EnvironmentVariablesFilterToggleShowSpecOnly] : ControlsStrings.EnvironmentVariablesFilterToggleShowAll)"
                           aria-label="@(_showAll ? Loc[ControlsStrings.EnvironmentVariablesFilterToggleShowSpecOnly] : ControlsStrings.EnvironmentVariablesFilterToggleShowAll)"
```

Comment:
I wonder if we should change the title and aria label too? Indicate why it's disabled?

Reply:
Can do. What text would you suggest? We'd go from two to four possible strings for each.

Or we just go back to hiding it :)

Reply:
I think you'd only have three strings. When disabled, say "No environment variables from[/provided by?] spec" or maybe "Spec provided no environment variables". When not disabled, use the existing two as appropriate. Probably makes sense to pull that logic out into a property at that point given the duplication and the extra layer of conditional.

I'm now imagining people reading the above suggestions and thinking "well why not?"... but I suppose that's a documentation issue. I'm already not confident users will know what spec means immediately, but that's in the original strings too.

Reply:
Thanks. I think the description is more like "All environment variables here were present in the spec, so there are no extras to show/hide."

Reply:
Gah, yeah, I had the condition reversed in my head (thinking there'd always be non-spec vars at runtime). Perhaps @leslierichardson95 wants to chime in on the language? I can't decide if the second half - effectively "so there is nothing to toggle" should be explicit or not.

But actually that brings something else up. Wouldn't the button also not do anything in the reverse case? If none of them are from spec? Is that even possible in Aspire since this mechanism is used to pass so much info around?

Reply:
In the reverse case, nothing would be visible until you toggled it.

+1 for Leslie's wordsmithery!

Reply:
@tlmii any further thoughts on this PR? What would you need to see in order to get this merged?

Reply:
> What would you need to see in order to get this merged

A friendly ping, apparently 😊 This one got lost amongst all the (positive!) churn in my notifications I think.

I'm fine with how this is for now. I think we should just keep an eye on if the reverse scenario is even possible and if we need more description of why it is disabled. But that can be a separate PR rather than letting this one get stale waiting.